### PR TITLE
290 resampling tests

### LINKED ISSUE #290: Write down- and resampling frontend tests
Put special attention on the trialdefinition rescaling (`trl /= factor`), as this might break things (with extra columns especially).

### PR DESCRIPTION
**Pull request by tensionhead**
## `resampledata` frontend tests

closes #290 

Author Guidelines
-----------------
- [x] Is the change set **< 600 lines**?
- [x] Was the code checked for memory leaks/performance bottlenecks?
- [x] Is the code running locally **and** on the ESI cluster?
- [x] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do **parallel** loops have a set length and correct termination conditions?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [x] Code layout
  - [x] Is the code PEP8 compliant?
  - [x] Does the code adhere to agreed-upon naming conventions?
  - [x] Are keywords clearly named and easy to understand?
  - [x] No commented-out code?
- [x] Are all docstrings complete and accurate?
- [x] Is the CHANGELOG.md up to date?

### CODE REVIEW

File: syncopy/preproc/resampledata.py [hunk 1]
```diff
@@ -41,16 +41,15 @@ def resampledata(data,
...
         The new sampling rate `resamplefs` must be an integer division of
         the old sampling rate, e. g., 500Hz to 250Hz.
-        Note that no anti-aliasing filtering is performed before downsampling,
+        NOTE: No anti-aliasing filtering is performed before downsampling,
         it is strongly recommended to apply a low-pass filter
-        beforehand via :func:`~syncopy.preprocessing` with the new
-        Nyquist frequency (`resamplefs / 2`) as cut-off.
-        Alternatively explicitly setting `lpfreq` (and `order` if needed)
-        performs anti-aliasing on the fly.
+        via explicitly setting `lpfreq` to the new Nyquist frequency
```
Comment (dfsp-spirit) [inline syncopy/preproc/resampledata.py:46]: Reads way better this way, and suggests the easiest solution first. :)

File: syncopy/preproc/resampledata.py [hunk 2]
```diff
@@ -195,18 +203,15 @@ def resampledata(data,
         filtered = AnalogData(dimord=data.dimord)
         AntiAliasFilter.initialize(
             data,
-            filtered._stackingDimm,
+            filtered._stackingDim,
```
Comment (dfsp-spirit) [inline syncopy/preproc/resampledata.py:206]: nice catch

Comment (tensionhead) [reply syncopy/preproc/resampledata.py:206]: Got catched with a `NameError` :grinning:

File: syncopy/tests/test_resampledata.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        else:
+            return spec_ds
+
+    def test_aa_filter(self):
+
+        # filter with new Nyquist
+        kwargs = {'resamplefs': self.fs // 2,
+                  'lpfreq': self.fs // 4}
+
+        spec_ds = self.test_downsampling(power_fac=1, **kwargs)
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/tests/test_resampledata.py:89]: uuh, true forgot to remove that guy.. :+1:

**Pull request merged**